pkg/autoscaling/handler.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoscaling
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/v4/server"
+	"github.com/unrolled/render"
+)
+
+// Strategy within a HTTP request provides rules and resources to help make decision for auto scaling.
+type Strategy struct {
+	Rules     []*Rule     `json:"rules"`
+	Resources []*Resource `json:"resources"`
+}
+
+// Rule is a set of constraints for a kind of component.
+type Rule struct {
+	Component               string       `json:"component"`
+	CPURule                 *CPURule     `json:"cpu_rule"`
+	StorageRule             *StorageRule `json:"storage_rule"`
+	ScaleOutIntervalSeconds uint64       `json:"scale_out_interval_seconds"`
+	ScaleInIntervalSeconds  uint64       `json:"scale_in_interval_seconds"`
+}
+
+// CPURule is the constraints about CPU.
+type CPURule struct {
+	MaxThreshold  float64  `json:"max_threshold"`
+	MinThreshold  float64  `json:"min_threshold"`
+	MaxCount      uint64   `json:"max_count"`
+	ResourceTypes []string `json:"resource_types"`
+}
+
+// StorageRule is the constraints about storage.
+type StorageRule struct {
+	MaxThreshold  float64  `json:"max_threshold"`
```

Comment:
I think we should use min instead of max.

Reply:
Can you explain more why we need to replace max with min?

Reply:
What does storage threshold mean? The ratio of used space to total space or available space to total space?

Reply:
The ratio of used space to total space. And I will add more comments if this API has no more problems.

Reply:
https://github.com/pingcap/tidb-operator/blob/master/pkg/autoscaler/autoscaler/calculate/storage.go#L75
TiDB operator is using available size to determine whether under storage pressure

Reply:
> The ratio of used space to total space. And I will add more comments if this API has no more problems.

We can't use used space and total space because we can't guarantee the target storage have its own volume. Maybe there exist other storage systems share the same volume with us.